ultravox/data/datasets.py

```diff
@@ -966,8 +1029,11 @@ def create_dataset(name: str, args: VoiceDatasetArgs) -> data.IterableDataset:
         "soda": SodaDataset,
         "dummy": LibriSpeechDummyDataset,
     }
-    name, *ext = name.split(":")
-    return DATASET_MAP[name](args, *ext)
+    if isinstance(name, config_base.DataDictConfig):
+        return GenericVoiceDataset(args, name)
```

Comment:
I don't like the idea of having a "special" class that works completely different from the rest of the classes and has its own special input format. When you go down this path soon you're gonna have 2, 3, more special classes.

Reply:
A majority of existing dataset classes can be handled by this generic dataset class. I think it makes sense to have a predefined dataset class for the datasets that we use frequently. At the same time, this could also be achieved using a predefined config (handled by a generic dataset class) as part of the code repo.

Reply:
I'm not talking about functionality here (there I do think the generic class is lacking, but it can improve).

I'm talking about code structure from a purely software-engineering stand point.

IMO if you want to change the structure, you should change it once and for all, not have special cases. We don't have to preserve backward compatibility much.

Reply:
Agreed. Maybe we keep this feature off the main for now until we come up with something nicer.

Reply:
Just to be clear, I'm suggesting maybe we could change the structure for all datasets, but just keep it consistent. And the special cases get handled in one place instead of two (once when configs are loaded and once in `create_datasets`).

Reply:
Agreed. That would require some time to iron out all the details and corner cases.